cli/azd/internal/repository/app_init.go

```diff
@@ -429,7 +449,103 @@ func prjConfigFromDetect(
 			name = config.Name
 		}
 		name = names.LabelName(name)
+		svc.Name = name
 		config.Services[name] = &svc
+
+		svcMapping[prj.Path] = name
+	}
+
+	if addResources {
+		config.Resources = map[string]*project.ResourceConfig{}
+		dbNames := map[appdetect.DatabaseDep]string{}
+
+		for database := range detect.Databases {
+			if database == appdetect.DbRedis {
+				redis := project.ResourceConfig{
+					Type: project.ResourceTypeDbRedis,
+					Name: "redis",
+				}
+				config.Resources[redis.Name] = &redis
+				dbNames[database] = redis.Name
+				continue
+			}
+
+			var dbType project.ResourceType
+			switch database {
+			case appdetect.DbMongo:
+				dbType = project.ResourceTypeDbMongo
+			case appdetect.DbPostgres:
+				dbType = project.ResourceTypeDbPostgres
+			}
+
+			db := project.ResourceConfig{
+				Type: dbType,
+			}
+
+			for {
+				dbName, err := promptDbName(i.console, ctx, database)
+				if err != nil {
+					return config, err
+				}
+
+				if dbName == "" {
+					i.console.Message(ctx, "Database name is required.")
+					continue
+				}
+
+				db.Name = dbName
+				break
+			}
```

Comment:
please move this to a new method to help with indentation and readability.  Since the context is already a for loop, it is easy to confuse a reader about the existing of a new `for`, taking extra time to recognize the effects of `continue` and `break`.

Reply:
It is written this way so that we can more easily merge the alpha vs. non-alpha code cleanly, once we remove the backcompat `alpha` feature. The code is slightly ugly here so that we can move quicker and cleaner in the future -- I hope that is acceptable in your mind.

Reply:
Help me understand how's the alpha switch related to having a for loop inside another for loop and having `break` and `continue` for both loops

Reply:
Certainly.

The logic 

```golang
for {
				dbName, err := promptDbName(i.console, ctx, database)
				if err != nil {
					return config, err
				}

				if dbName == "" {
					i.console.Message(ctx, "Database name is required.")
					continue
				}

				db.Name = dbName
				break
			}
```

would look like

```golang
dbName, err := promptDbName(i.console, ctx, database)
if err != nil {
   return config, err
}
```

in the future, once we merge the name validation into the `promptDbName` prompt directly. The reason why it doesn't work today, is that easy init requires the `promptDbName` functionality to not validate for an empty name.